blog/views.py

```diff
@@ -0,0 +1,46 @@
+from django.http import HttpResponse, HttpRequest, HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views import generic
+
+from blog.forms import CommentaryForm
+from blog.models import Post
+
+
+class PostListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.select_related("owner").prefetch_related(
+        "commentaries"
+    )
+    paginate_by = 5
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+    def get_object(self, queryset=None) -> Post:
+        return Post.objects.prefetch_related("commentaries__user").get(
+            id=self.kwargs["pk"]
+        )
+
+    def get_context_data(self, **kwargs) -> dict:
+        context = super().get_context_data(**kwargs)
+        context["form"] = CommentaryForm
```

Comment:
In the 'get_context_data' method, you are passing the CommentaryForm class itself to the context, not an instance of it. This could lead to unexpected behavior when trying to render the form in the template. You should create an instance of the form and pass that to the context instead.